File: homestay-backend/Homestay.Api/Domain/Entities/Homestay.cs
using Homestay.Api.Domain.Common;

namespace Homestay.Api.Domain.Entities;

public partial class Homestay : ISoftDeletable
{
    public Guid Id { get; set; }

    public Guid HostId { get; set; }

    public string Name { get; set; } = null!;

    public string Address { get; set; } = null!;

    public string City { get; set; } = null!;

    public string? Description { get; set; }

    public DateTime? CreatedAt { get; set; }
    
    public DateTime? DeletedAt { get; set; }

    public Guid? DeletedBy { get; set; }

    public virtual User Host { get; set; } = null!;

    public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();

    public virtual ICollection<Room> Rooms { get; set; } = new List<Room>();
}

File: homestay-frontend/types/homestay.ts
export interface Homestay {
  id: string;
  name: string;
  description?: string;
  address: string;
  city: string;
  hostId: string;
  hostName?: string;
  createdAt: string;
}

export interface CreateHomestayRequest {
  name: string;
  description?: string;
  address: string;
  city: string;
}

export interface UpdateHomestayRequest {
  name?: string;
  description?: string;
  address?: string;
  city?: string;
}

export interface HomestayResponse {
  id: string;
  name: string;
  description?: string;
  address: string;
  city: string;
  hostId: string;
  createdAt: string;
}
